astroplan/core.py

```diff
@@ -1114,10 +1115,65 @@ def moon_illumination(self, time):
             the `~astropy.time.Time` initializer, so it can be anything that
             `~astropy.time.Time` will accept (including a `~astropy.time.Time`
             object).
+
+        moon : `~astropy.coordinates.SkyCoord` or `None` (default)
```

Comment:
I don't quite understand why here (and in `moon_phase`) you have the `moon` and `sun` keywords.  Doesn't it make more sense to just use `time`?  Otherwise the user might pass in `moon`/`sun` objects that don't correctly match `time`, and then all kinds of stuff might go bonkers (without an error).  After all, one of the Zen of python elements is "Errors should never pass silently."...

If this is a performance thing, I would instead suggest caching.  I.e. something like:
```
# somewhere you'll have to create self._cached_moons... in __init__ is probably the easiest
if time in self._cached_moons:  # this might need to be time.jd? Not sure if `Time` objects are hashable
    moon = self._cached_moons[time]
else:
    moon = get_moon(time, self.location, self.pressure)
    self._cached_moons[time] = moon
# similar for the sun
```